<?xml version="1.0"?>
<launch>

    <!--<node pkg="rosbag" type="play" name="rosbag_play" args="--clock -r 1 /media/jquenzel/CopterExchange/ul_hk/20190331_HH_lidar.bag"/>-->
    <!--<node pkg="rosbag" type="play" name="rosbag_play" args="--clock -r 1 /media/jquenzel/CopterExchange/ul_hk/20190331_NJ_LL_lidar.bag"/>-->
    <!--<node pkg="rosbag" type="play" name="rosbag_play" args="--clock -r 1 /media/jquenzel/CopterExchange/ul_hk/20190331_NJ_SL_lidar.bag"/>-->
    <!--<node pkg="rosbag" type="play" name="rosbag_play" args="--clock -r 1 /media/jquenzel/CopterExchange/ul_hk/20190331_WH_lidar.bag"/>-->
    <!--<param name="scan_topic" value="/velodyne_points" />-->

    <param name="scan_topic" value="/velodyne_points_0" />
    <!--<node pkg="rosbag" type="play" name="rosbag_play" args="--clock -r 1 /media/jquenzel/CopterExchange/ul_hk/test2.bag"/>-->
    <!--<node pkg="rosbag" type="play" name="rosbag_play" args="--clock -r 1 /media/jquenzel/CopterExchange/ul_hk/test3.bag"/>-->
    <node pkg="rosbag" type="play" name="rosbag_play" args="--clock -r 1 /media/jquenzel/CopterExchange/ul_hk/urban_loco_hk_5.bag"/>

    
    <!-- For Velodyne VLP-16 
    <param name="scan_line" value="16" />
    -->

    <!-- For Velodyne HDL-32 -->
    <param name="scan_line" value="32" />

    <!-- For Velodyne HDL-64 
    <param name="scan_line" value="64" />
    -->
    
    <!-- For Ouster OS-0 -->
    <!--param name="scan_line" value="128" /-->
    
    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />
    <param name="scan_period" value="0.1" />

    <param name="vertical_angle" type="double" value="2.0" />
    <param name="max_dis" type="double" value="100.0" />
    <param name="map_resolution" type="double" value="0.4" />
    <param name="min_dis" type="double" value="1.0" />
    <!--- -->
    <node pkg="floam" type="floam_odom_estimation_node" name="floam_odom_estimation_node" output="screen"/>
    <node pkg="floam" type="floam_laser_processing_node" name="floam_laser_processing_node" output="screen"/>
    <node pkg="floam" type="floam_laser_mapping_node" name="floam_laser_mapping_node" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 0 0 0 /world /map 10" />
    
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find floam)/rviz/floam_mapping.rviz" />
    </group>

    
  	<node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="gt" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="velodyne" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>
    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="base_link" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="base_link" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
